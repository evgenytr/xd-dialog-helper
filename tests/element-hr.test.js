const hr = require('../dialog-helper').types.HR;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {}),
    close: jest.fn(() => {}),
    cancel: jest.fn(() => {}),
    registerElement: jest.fn(() => undefined)
};

describe('HR Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            const rendered = hr.render('dialog', {
                id: 'hr'
            }, actions);
            expect(rendered).toMatchSnapshot();
        });
    });
    describe('value()', () => {
        it('should correctly return undefined as value', () => {
            expect(hr.value({wrapper: document.createElement('hr')})).toBeUndefined();
        });
    });
    describe('validate()', () => {
        it('should always return true', () => {
            expect(hr.valid({})).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(hr.type).toMatchSnapshot();
    });
});
