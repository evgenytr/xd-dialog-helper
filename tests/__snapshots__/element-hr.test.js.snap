// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Horizontal Rule"`;

exports[`HR Element render() should render correctly 1`] = `
Object {
  "input": undefined,
  "props": Object {
    "id": "hr",
  },
  "type": Object {
    "render": [Function],
    "type": "Horizontal Rule",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <hr
    id="dialog-hr"
  />,
}
`;
